### DISCUSSION TITLE
usb.core.NoBackendError: No backend available

### DISCUSSION THREAD

**Discussion by simonwanghub** [pyusb/pyusb#499] [Q&A] [answered]
<img width="1440" alt="image" src="https://github.com/pyusb/pyusb/assets/30173054/7e46b523-93cd-4780-84f9-c321996d1159">

On Windows 10, I can't find the apple device in recovery mode (iboot).

**Comment by mcuee** [comment, marked as answer]
Make sure you install the supported driver like WinUSB. You may want to do that through Zadig which will automatically install libusb0.dll for you to the system. From there you can use the libusb-0.1 backend.

But it is even better to have libusb-1.0.dll in the system so that you can use that backend.

Zadig: https://github.com/pbatard/libwdi/releases/tag/v1.5.0
libusb-1.0 download: https://github.com/libusb/libusb/releases/tag/v1.0.26

**Comment by rrakso** [comment]
On Ubuntu all I had to do was `sudo apt install libusb-1.0-0` 😉
